### PR TITLE
WIP - istiod safe mode

### PR DESCRIPTION
**Pull request by keithmattix**
**Please provide a description of this PR:**
Add a new WIP helm chart for istiod's "safe mode" whose values.yaml will be a strict subset of the options in the main istiod Helm chart. The proposal doc for Istio "safe mode" can be found [here](https://docs.google.com/document/d/1aaORW2Ak4Vfpr-N68Q04qS7iskDdF3v7lcZFQsFA_L0/edit#).

### CODE REVIEW

**Review by costinm — APPROVED** [review]
I would move this to manifests/charts/istiod-safe - ztunnel is also top level, and the 'istio discovery' and 'istio control' names are ancient and no longer very representative - we use "Istiod" as service and deployment names, etc.

Other than that - I would merge this as is, reflecting the current state - before making additional changes that remove or set different defaults, so we have a good history.

I am approving since it's (I hope) same code we have - but we should wait until the doc is discussed in the WG.

File: manifests/charts/istio-control/istio-discovery-safe/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+name: istiod
```
Comment (keithmattix) [inline manifests/charts/istio-control/istio-discovery-safe/Chart.yaml:2]: Blocking myself until I at least change the name of this chart

File: manifests/charts/istio-control/istio-discovery-safe/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (bleggett) [inline manifests/charts/istio-control/istio-discovery-safe/kustomization.yaml:1]: Can we de-Kustomize?

Using Kustomize and Helm together seems redundant.

Comment (costinm) [reply manifests/charts/istio-control/istio-discovery-safe/kustomization.yaml:1]: +1 

However IMO we should first make an exact copy of the original chart, and start removing/changing in separate PRs, so there is a clear history of what is different and we can review changes. 

For initial PR we just need to agree on the name - I don't like 'istio-discovery-safe' - we can just add manicests/charts/istiod-safe ( and rename later ).

The istio-control/istio-discovery is an artifact from the ancient days when we had many separate charts before istiod.

Comment (keithmattix) [reply manifests/charts/istio-control/istio-discovery-safe/kustomization.yaml:1]: +1 to Costin; I'm all for removing kustomize; but we should prioritize a clear diff history where we can.

**Comment by costinm** [comment]
Any progress on this ? It seems we settled on 'foundation' - can we start with the exact copy of istio-discovery so we can track changes, and get the ball rolling ? We'll have a lot of work to cleanup that we can split - I don't think there is any major objection on starting the process, but individual changes and the intial release and surface may be discussed later.

**Comment by keithmattix** [comment]
Based on feedback we got from the community, I think it probably makes sense to shelf this effort for now until we are more confident in our feature hygiene. More to come in that area in the next couple of weeks

**Comment by costinm** [comment]
I don't know what you mean by 'feature hygiene' or 'confident' - but I
think having a clean chart with no experimental or alpha options is still a
valuable effort
and orthogonal to any future hygiene.

**Comment by keithmattix** [comment]
Based on today's conversation in the community meeting, I'm closing this PR

**Pull request closed without merging**